readme.md

```diff
@@ -45,11 +45,16 @@ Type: `object`
 
 ##### deep
 
-Type: `boolean`<br>
+Type: `boolean | Function`<br>
 Default: `false`
 
 Recurse nested objects and objects in arrays.
 
+If using a boolean value it applies to all keys in all nested objects. If using a `DeepKeyFilter`
+
+- It has signature `deep(sourceKey)`.
+- It must return a single boolean value: `true|false` if the nested object belonging to `sourceKey` should be mapped
```

Comment:
I think this added text here could have been written clearer. It should start with what exactly `DeepKeyFilter` does and what problems it solves.

The readme docs should also be in sync with the index.d.ts docs.

Reply:
Dear @sindresorhus! I was unfortunately a bit busy with other projects. Now I want to come back to our recent discussions. I extended the Readme-file also with an example to better explain what the DeepKeyFilter is doing and also synced the index.d.ts file. Hope you'll like it.

Looking forward to your opinion. Cheers!

Reply:
Resolve, because outdated